### PR TITLE
Automate stripe-mock update

### PR DESCRIPTION
**Pull request by pakrym-stripe**
Adds a workflow to update stripe-mock on openapi updates.

Uses my fork of stripe-mock for testing, I'll switch to the main repo after testing is done.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    - run: |
+        rm -f ./stripe-mock/openapi/spec3.json
+        rm -f ./stripe-mock/openapi/fixtures3.json
+        cp openapi/spec3.json stripe-mock/openapi/spec3.json
+        cp openapi/fixtures3.json stripe-mock/openapi/fixtures3.json
+    - name: Create Pull Request
+      uses: peter-evans/create-pull-request@v4
+      with:
+        path: stripe-mock
+        token: ${{ secrets.BOT_USER_PAT }}
```
Comment (dcr-stripe) [inline .github/workflows/publish.yml:28]: Nit: Is this a name we control? Let's make this more explicit - acronyms are easy to forget/get confused :)

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        path: stripe-mock
+    - run: |
+        rm -f ./stripe-mock/openapi/spec3.json
+        rm -f ./stripe-mock/openapi/fixtures3.json
+        cp openapi/spec3.json stripe-mock/openapi/spec3.json
+        cp openapi/fixtures3.json stripe-mock/openapi/fixtures3.json
+    - name: Create Pull Request
+      uses: peter-evans/create-pull-request@v4
+      with:
+        path: stripe-mock
```
Comment (dcr-stripe) [inline .github/workflows/publish.yml:27]: Assuming here we'll add title/body/reviewers after this looks good?

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: Publish
+
+on:
+  workflow_dispatch: {}
+  push:
```
Comment (dcr-stripe) [inline .github/workflows/publish.yml:5]: Nit: Let's start with just workflow_dispatch while we're iterating on this?

**Review by dcr-stripe — APPROVED** [review]
LGTM. Very small nits.

Do you want to do this in a branch rather than on master? Might help you be able to iterate faster if you need tweaking.

File: .github/workflows/publish.yml (hunk 2 above)
Comment (pakrym-stripe) [reply .github/workflows/publish.yml:27]: Correct.

**Comment by pakrym-stripe** [comment]
>Do you want to do this in a branch rather than on master? Might help you be able to iterate faster if you need tweaking.

I think it has to be in master so it can be triggered. I can trigger it on the branch.

**Pull request merged**